File: design-patterns/go/factory/transport-method/bike.go
// bike.go

package main

import "fmt"

type Bike struct {
}

func NewBike() (bike *Bike) {
	bike = &Bike{}
	return
}

func (bike *Bike) Repair() {
	fmt.Println("Bike Repair")
}

func (bike *Bike) Start() {
	fmt.Println("Bike started")
}

func (bike *Bike) Stop() {
	fmt.Println("Bike Stopped")
}

File: design-patterns/php/factory/transport-method/Bike.php
<?php
// Bike.php

namespace Factory\TransportMethod;

class Bike implements Transport {
    public function start() {
        echo "Bike started\n";
    }

    public function stop() {
        echo "Bike Stopped\n";
    }

    public function repair() {
        echo "Bike Repair\n";
    }
}